ddcz/views.py

```diff
@@ -291,7 +293,16 @@ def author_detail(request, author_id, slug):
         'pages_with_creations': author.get_all_creations(),
     })
 
+
 def phorum(request):
+    if request.method == "POST":        
+        form = PhorumCommentForm(request.POST)
+        user = get_object_or_404(UserProfile, nick_uzivatele=request.user)
```

Comment:
OK, so you don't h ave the security hole, this is correct. But why do you post the username in form then?

Reply:
Also, there should a guard: what if unauthenticated user makes a POST request?

Reply:
I though that get_object_or_404 will throw 404 if the request.user will be null/false. 
``` 
except queryset.model.DoesNotExist:
        raise Http404(
            "No %s matches the given query." % queryset.model._meta.object_name
        )
```
What are my other possibilities then? Something like `if request.user:`?

Reply:
yes, `if request.user` — this will cause issues should there be accidentally userProfile with nick null :) 

But I am intrigued this works as I thought `request.user` contains `User` object, whereas `nick_uzivatele` asks for string...